### PR TITLE
Update calendar_v3.d.ts

### PR DESCRIPTION
**Pull request by tdesrosi**
Add support for Working Locations in Calendar.Event, recently added to the API last month.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change. 
    - **Calendar v3 API does not seem to be already covered by tests.**
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [calendar v3 API reference](https://developers.google.com/calendar/api/v3/reference/events#resource)
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header. 
    - **N/A** -- The "Version" didn't change (v3). This brings the event object up to date with releasing of working location data to the events resource

### CODE REVIEW

**Comment by tdesrosi** [comment]
Updating WorkingLocationProperties with homeOffice: unknown|undefined.

The documentation does not provide a specific type: 

workingLocationProperties.homeOffice: any value: If present, specifies that the user is working at home.

File: types/google-apps-script/apis/calendar_v3.d.ts
Comment (mtgto) [inline types/google-apps-script/apis/calendar_v3.d.ts:524]:
```suggestion
      interface EventWorkingLocationPropertiesOfficeLocation {
        buildingId?: string | undefined;
        floorId?: string | undefined;
        floorSectionId?: string | undefined;
        deskId?: string | undefined;
        label?: string | undefined;
      }
      interface EventWorkingLocationPropertiesCustomLocation {
        label: string;
      }
      interface EventWorkingLocationProperties {
        type?: string | undefined;
        homeOffice?: object | undefined;
        customLocation?: EventWorkingLocationPropertiesCustomLocation | undefined;
        officeLocation?: EventWorkingLocationPropertiesOfficeLocation | undefined;
      }
```

1. Add space between `|`.
2. Add "Event" prefix to interface names. It is same type names written in type hinting of Google Apps Script Editor.
    1. For example, you can find the function `Calendar.newEventWorkingLocationProperties()` exists. It returns EventWorkingLocationProperties.
    2. `Calendar.newEventWorkingLocationPropertiesOfficeLocation()` and `Calendar.newEventWorkingLocationPropertiesCustomLocation()` also exist.

File: types/google-apps-script/apis/calendar_v3.d.ts
Comment (mtgto) [inline types/google-apps-script/apis/calendar_v3.d.ts:391]:
```suggestion
        workingLocationProperties?: Calendar.Schema.EventWorkingLocationProperties | undefined;
```

File: types/google-apps-script/apis/calendar_v3.d.ts
Comment (mtgto) [inline types/google-apps-script/apis/calendar_v3.d.ts:527]: Could you add new three functions to interaface Calendar?

```ts
newEventWorkingLocationProperties(): Calendar.Schema.EventWorkingLocationProperties;
newEventWorkingLocationPropertiesCustomLocation(): Calendar.Schema.EventWorkingLocationPropertiesCustomLocation;
newEventWorkingLocationPropertiesOfficeLocation(): Calendar.Schema.EventWorkingLocationPropertiesOfficeLocation;
```

**Review by mtgto — CHANGES_REQUESTED** [review]
Thank you for your contribution.

1. Please fix nits because this type definition is automatically generated by the tool using type hinting in google apps script.
2. Can you update test/google-apps-script-tests.ts to use workingLocationProperties? It reduces review time.
    - You can find `Untested Change` label is added by bot. It means that this pull request requires repo maintainer approval https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66705#issuecomment-1719910950

example:
```ts
let event: GoogleAppsScript.Calendar.Schema.Event = {
  summary: 'Lunch Meeting',
  ...
  workingLocationProperties: {
      type: "homeOffice",
      homeOffice: { "foo": "bar" },
  }
};
```
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/93480a775a6b0181fc0cf2ec5e3b5002d70fcaec/types/google-apps-script/test/google-apps-script-tests.ts#L52-L66

**Comment by tdesrosi** [comment]
Thanks for the review! My apologies on the tests, I didn't realize it was one suite for all GAS typings. Please let me know if there is anything else to add.

File: types/google-apps-script/test/google-apps-script-tests.ts
Comment (mtgto) [inline types/google-apps-script/test/google-apps-script-tests.ts:104]: (You need not to fix it) My guess is add `workingLocationProperties` prop to above `event` defined in L52.
But nothing wrong with this. Thank you for adding test.

**Review by mtgto — APPROVED** [review]
Looks good to me. Thank you for your contribution again.

> My apologies on the tests, I didn't realize it was one suite for all GAS typings.

I hope you don't mind. This repository is huge and manages under some rules. But it's hard to understand.

**Comment by tdesrosi** [comment]
Ready to merge

**Pull request merged**
